### ISSUE TITLE
PHP 7.4 issues with cacti 1.2.7

### ISSUE THREAD

**Issue by mortenstevens** [Cacti/cacti#3127]
Just a quick test on Fedora rawhide with PHP 7.4 and cacti 1.2.7:

2019/12/02 16:16:03 - CMDPHP PHP ERROR NOTICE Backtrace: (CactiShutdownHandler(), /lib/functions.php[4558]:CactiErrorHandler())
--
2019/12/02 16:16:03 - ERROR PHP NOTICE: Trying to access array offset on value of type null in file: /usr/share/cacti/lib/functions.php on line: 4558
2019/12/02 16:16:03 - CMDPHP PHP ERROR NOTICE Backtrace: (CactiShutdownHandler(), /lib/functions.php[4554]:CactiErrorHandler())
2019/12/02 16:16:03 - ERROR PHP NOTICE: Trying to access array offset on value of type null in file: /usr/share/cacti/lib/functions.php on line: 4554

Best regards,

Morten

**Comment by cigamit** [comment]
@mortenstevens, is this against 1.2.x or 1.2.7?

**Comment by cigamit** [comment]
Okay, confirming that this was 1.2.7.  This has already been resolved.

**Comment by cigamit** [comment]
Duplicate of #2130

**Comment by mortenstevens** [comment]
@cigamit 
Are you sure that this is related to #2130?

**Comment by cigamit** [comment]
Yea, totally.  The line number from a raw 1.2.7 install lined up 100% with the other issue.

**Comment by athompso** [comment]
I'm experiencing this same error message on the Debian-packaged version of Cacti 1.2.2 as soon as I upgraded to PHP 7.4, but in functions.php line 4424.
Any idea what commit fixed the issue in 1.2.7, so I can manually backport it into this version?  I don't see any commits linked to #2130.
The error in question is referring to `$error = ... $phperrors[$level] ...` in CactiErrorHandler.
I've been unable to get a backtrace, so I've no idea who's calling CactiErrorHandler.
I've also browed the entire change history of function.php between 1.2.2-release and 1.2.7-release without seeing anything that would fix this...

```versions
cacti          1.2.2+ds1-2+deb10u1
php7.4-fpm     7.4.1-1+0~20191218.8+debian10~1.gbp21c50e
```

Thoughts/ideas?
